api/opencl_architecture.asciidoc

```diff
@@ -2104,6 +2104,20 @@ The new version of OpenCL C must be explicitly requested via the `-cl-std=` buil
 Whenever an OpenCL C feature is optional in the new version of the OpenCL C programming language, it will be paired with a feature macro, such as `+__opencl_c_feature_name+`, and a corresponding API query.
 If a feature macro is defined then the feature is supported by the OpenCL C compiler, otherwise the optional feature is not supported.
 
+In order to allow future versions of OpenCL to support new types of
+devices, some capabilities that are currently required of all OpenCL
```

Comment:
The paragraphs above this one use the term "feature" so I'd be inclined to continue using that.   Unless you're trying to stay away from the specific features made optional in OpenCL 3.0?